Controllers/User_test.go

```diff
@@ -1,42 +1,193 @@
-package Controllers
+package controller
 
 import (
-	"first-api/Models"
+	"bytes"
+	"encoding/json"
+	model "first-api/Models"
+	"net/http"
 	"testing"
+
 	"github.com/stretchr/testify/assert"
-)
 
+	"net/http/httptest"
+
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/mock"
+)
 
+func TestValidateUserData(t *testing.T) {
 
-func TestValidateUserData(t *testing.T){
-	assert  := assert.New(t)
+	assert := assert.New(t)
 
-	testService := Models.User{Name:"",Email:"test@gmail.com",Phone:"9999999999",Address:"abcd efgh ijkl"}
+	testService := model.User{Name: "", Email: "test@gmail.com", Phone: "9999999999", Address: "abcd efgh ijkl"}
 
-	err := testService.Validate();
+	err := testService.Validate()
 	// fmt.Println(err)
 
-	assert.NotNil(t,err);
+	assert.NotNil(t, err)
 
 	testService.Name = "testing"
 	testService.Email = "test"
-	err = testService.Validate();
+	err = testService.Validate()
 	// fmt.Println(err)
 
-	assert.NotNil(t,err);
-
+	assert.NotNil(t, err)
 
-	testService.Phone = "999999999" 
-	err = testService.Validate();
+	testService.Phone = "999999999"
+	err = testService.Validate()
 	// fmt.Println(err)
 
-	assert.NotNil(t,err);
+	assert.NotNil(t, err)
 
 	testService.Address = ""
 	err = testService.Validate()
 	// fmt.Println(err)
 
-	assert.NotNil(t,err);
+	assert.NotNil(t, err)
+
+}
+
+type MockUserStore struct {
+	mock.Mock
+}
+
+func (m *MockUserStore) CreateUser(user *model.User) error {
+	args := m.Called(user)
+	return args.Error(0)
+}
+
+func (m *MockUserStore) Validate(user model.User) error {
+	args := m.Called(user)
+	return args.Error(0)
+}
+
+func (m *MockUserStore) GetAllUsers(users *[]model.User) error {
+	args := m.Called(users)
+	return args.Error(0)
+}
+
+func (m *MockUserStore) GetUserByID(user *model.User, id string) error {
+	args := m.Called(user, id)
+
+	// if the mock is set to return an error, return it
+	if args.Error(0) != nil {
+		return args.Error(0)
+	}
+
+	// otherwise, set the passed user object fields
+	user.Id = 1
+	user.Name = "test user"
+	user.Email = "test@gmail.com"
+	user.Phone = "9999999999"
```

Comment:
move this implementation to Models folder under "Mock" folder , so that it could be reused by other tests as well.